practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java

```diff
@@ -126,15 +128,12 @@ public PractitionerDetails getPractitionerDetails(
                 practitionerIdStringValue =
                         practitioner.getIdElement() != null
                                         && practitioner.getIdElement().getIdPart() != null
-                                ? ((IBaseResource) practitioner)
-                                        .getIdElement()
-                                        .getIdPart()
-                                        .toString()
+                                ? practitioner.getIdElement().getIdPart()
                                 : "";
```

Comment:
We already have the null check in the outer condition as well as the default of `""`, we can refactor this line to just be   `practitionerIdStringValue = practitioner.getIdElement().getIdPart();`

Reply:
Makes sense, actually, I don't think we need to care about whether the id is a string or numeric.  https://github.com/opensrp/hapi-fhir/blob/148cbf119b3a90d129850ff19fc39ae3fdddc6c0/hapi-fhir-base/src/main/java/org/hl7/fhir/instance/model/api/IIdType.java#L50 returns a string which will work on search so we do not need to check whether the id is numeric or not